File: chattools/src/main/java/aytackydln/chattools/telegram/dto/response/DeleteMessage.java
package aytackydln.chattools.telegram.dto.response;

import aytackydln.chattools.telegram.exception.TelegramException;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.Data;

@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
@Data
public class DeleteMessage implements TelegramResponse {

    public static final String ERROR_WHILE_SENDING_DELETE_MESSAGE_RESPONSE_CHAT_ID_MESSAGE_ID
            = "Error while sending delete message response: \n chatId: {} messageId: {}";
    private final long chatId;
    private final long messageId;

    private boolean errorProcessed = false;

    @Override
    public String getMethod() {
        return "deleteMessage";
    }

    @Override
    @JsonIgnore
    public boolean isLimited() {
        return false;
    }

    @Override
    public TelegramResponse onError(TelegramException e) {
            if (!errorProcessed) {
                errorProcessed = true;
            }
        return null;
    }

    @Override
    public void preSend() {
        //doesn't need
    }
}

File: chattools/src/main/kotlin/aytackydln/chattools/telegram/dto/response/DeleteMessage.kt
package aytackydln.chattools.telegram.dto.response

import aytackydln.chattools.telegram.exception.TelegramException
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.databind.PropertyNamingStrategies
import com.fasterxml.jackson.databind.PropertyNamingStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming
import lombok.Data

@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
data class DeleteMessage(
    val chatId: Long = 0,
    val messageId: Long = 0,
) : TelegramResponse {

    private var errorProcessed = false

    override val method: String
        get() = "deleteMessage"

    @get:JsonIgnore
    override val isLimited: Boolean
        get() = false

    override fun onError(e: TelegramException): TelegramResponse? {
        if (!errorProcessed) {
            errorProcessed = true
        }
        return null
    }

    override fun preSend() {
        //doesn't need
    }

    companion object {
        const val ERROR_WHILE_SENDING_DELETE_MESSAGE_RESPONSE_CHAT_ID_MESSAGE_ID
                : String = "Error while sending delete message response: \n chatId: {} messageId: {}"
    }
}
